// Copyright 2023 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package google.cmrt.sdk.instance_database_client;

import "operator/protos/shared/backend/asginstance/instance_status.proto";
import "operator/protos/shared/backend/asginstance/asg_instance.proto";

option java_multiple_files = true;
option java_outer_classname = "InstanceDatabaseClientProto";
option java_package = "com.google.cmrt.sdk.instance_database_client";

// For faster allocations of sub-messages.
option cc_enable_arenas = true;

// Request to get the instance by name.
// In AWS, it is the instance ID.
// In GCP, it is of format:
// https://www.googleapis.com/compute/v1/projects/%s/zones/%s/instances/%s
message GetInstanceByNameRequest {
  // The instance name
  string instance_name = 1;
}

// Response of getting the instance by name.
message GetInstanceByNameResponse {
  // The instance object.
  google.scp.operators.protos.shared.backend.asginstance.AsgInstance instance =
      1;
}

// Request to list instances by status.
message ListInstancesByStatusRequest {
  // The instance status.
  google.scp.operators.protos.shared.backend.asginstance.InstanceStatus
      instance_status = 1;
}

// Response of listing instances by status.
message ListInstancesByStatusResponse {
  // The instance list.
  repeated google.scp.operators.protos.shared.backend.asginstance.AsgInstance
      instances = 1;
}

// Request to update instance.
message UpdateInstanceRequest {
  // The instance object to be updated. TTL will be ignored and not updated.
  google.scp.operators.protos.shared.backend.asginstance.AsgInstance instance =
      1;
}

// Response of updating instance.
message UpdateInstanceResponse {}
